File: C++/Funciones/isPrime.cpp
bool isPrime(int n){ //Función para determinar si un numero es primo.
    if (n == 1)return false;
    else if (n < 4)return true;
    else if (n % 2 == 0)return false;
    else if (n < 9)return true;
    else if (n % 3 == 0)return false;
    else{
		double r = sqrt(n); 
		int f = 5;         
        while (f <= r){
            if ((n % f) == 0) return false;
            else if (n % (f + 2) == 0) return false;
            else f += 6;                
        }return true;
    }return true;
}

File: Visual Basic .NET/Funciones/isPrime.vb
'Función usada en Project Euler #027
'Funcion para determinar si un numero es primo.

    Function isPrime(ByVal n As Integer) As Integer
        If n <= 0 Then
            Return False
        ElseIf n = 1 Then
            Return False
        ElseIf n < 4 Then
            Return True
        ElseIf n Mod 2 = 0 Then
            Return False
        ElseIf n < 9 Then
            Return True
        ElseIf n Mod 3 = 0 Then
            Return False
        Else
            Dim r As Integer = Math.Floor(Math.Sqrt(n))
            Dim f As Integer = 5
            Do While f <= r
                If n Mod f = 0 Then
                    Return False
                ElseIf n Mod (f + 2) = 0 Then
                    Return False
                Else
                    f += 6
                End If
            Loop
            Return True
        End If
    End Function
